src/themes/JSONTheme.ts

```diff
@@ -0,0 +1,1003 @@
+                "https://dl.dropboxusercontent.com/s/s079o0tm7ddmzr7/black_shade.png?dl=1"
+            ],
+            "blackShadows": [
+                "https://dl.dropboxusercontent.com/s/s079o0tm7ddmzr7/black_shade.png?dl=1"
+            ],
+            "sizes": [0.9],
+            "shadowSizes": [1.1]
+        }
+        `;
+
+        json = `
+        {
+            "name": "just-colors",
+            "boardColor": "#CBA170",
+            "boardInkColor": "#382933",
+            "whiteStoneColor": "#FAF1E8",
+            "blackStoneColor": "#2B2825",
+            "blackTextColor": "#FAF1E8",
+            "whiteTextColor": "#2B2825",
+            "blankTextColor": "#FAF1E8"
+        }
+        `;
+
+        json = ` 
+        {
+            "name": "shuffled-stones",
+            "boardColor": "#CBA170",
+            "boardInkColor": "#382933",
+            "whiteStoneColor": "#FAF1E8",
+            "whiteStoneLineWidth": 0,
+            "blackStoneColor": "#2B2825",
+            "blackTextColor": "#FAF1E8",
+            "whiteTextColor": "#2B2825",
+            "blankTextColor": "#FAF1E8",
+            "sizes": [0.98],
+            "offsets": [[0.02, 0.01], [0.05, -0.02], [-0.01, 0.05]]
+        }      
+        `;
+
+        json = `
+        {
+            "name": "Happy Stones",
+            "boardImage": "https://raw.githubusercontent.com/upsided/Upsided-Sabaki-Themes/main/happy-stones/goban_texture_fancy_orange.png",
+            "whiteStones": [
+                "https://raw.githubusercontent.com/upsided/Upsided-Sabaki-Themes/main/happy-stones/glass_white.png"
+            ],
+            "blackStones": [
+                "https://raw.githubusercontent.com/upsided/Upsided-Sabaki-Themes/main/happy-stones/glass_black.png"
+            ],
+            "sizes": [2],
+            "offsets": [[0.45,0.45]]
+        }
+        `;
+        json = `
+        {
+            "name": "hikaru",
+            "boardImage": "https://raw.githubusercontent.com/upsided/Upsided-Sabaki-Themes/main/hikaru/board.svg",
+            "boardInkColor": "rgb(76, 47, 0, 0.8)",
+            "whiteStones": [
+                "https://raw.githubusercontent.com/upsided/Upsided-OGS-Themes/main/ogs-hikaru/hikaru_white_stone_raw.svg"
+            ],
+            "blackStones": [
+                "https://raw.githubusercontent.com/upsided/Upsided-OGS-Themes/main/ogs-hikaru/hikaru_black_stone_raw.svg"
+            ],
+            "shadows": [
+                "https://raw.githubusercontent.com/upsided/Upsided-OGS-Themes/main/ogs-hikaru/hikaru_stone_shadow.svg"
+            ],
+            "shadowOffsets": [[0.02, 0.1]],
+            "shadowSizes": [1.1]
+        }
+        `;
+
+        return json;
+    }
+}
+
+export default function (GoThemes: GoThemesInterface) {
+    JSONTheme.setJSON(JSONTheme.getDefaultJSON());
+    GoThemes["black"]["JSONTheme"] = JSONTheme;
+    GoThemes["white"]["JSONTheme"] = JSONTheme;
+    GoThemes["board"]["JSONTheme"] = JSONTheme;
```

Comment:
Instead of using a static theme system, what about doing something like using a generator to create your instances of JSONTheme classes specific for whatever settings the user has provided? Something like

```
function JSONThemeGenerator(json: string): typeof JSONTheme {
    return class Theme extends JSONTheme {
        public json:any = JSON.parse(json);
    };
}
```

then

```
GoThemes["board"]["hikaru"] = JSONThemeGenerator(hikaru_json);
```
where `hikaru_json` would be that string blob from up above? You could then generate those dynamically and load them localStorage or whatever you wanted and we could avoid the headache of the static settings and whatnot?

Reply:
Yeah, I like this. I was afraid of integration with OGS but I'm pumped up enough to send a legit PR that implements it. If JSONTheme is instance-based, then the user can have all kinds of themes at their beck & call. Was thinking a "horizontal scroll" theme picker with a "+" on the end to add another...